lib/YaST/DNSServer/DNSServerSetupController.pm

```diff
@@ -0,0 +1,99 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: The class introduces actions for DNS Server Settings Page.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package YaST::DNSServer::DNSServerSetupController;
+use strict;
+use warnings;
+use YaST::DNSServer::ForwarderSettingsPage;
+use YaST::DNSServer::ZonesPage;
+use YaST::DNSServer::FinishWizardPage;
+use Installation::Popups::YesNoPopup;
+use YuiRestClient;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {}, $class;
+    return $self->init($args);
+}
+
+sub init {
+    my ($self, $args) = @_;
+    $self->{FWSettingsPage} = YaST::DNSServer::ForwarderSettingsPage->new({app => YuiRestClient::get_app()});
+    $self->{ZonesPage} = YaST::DNSServer::ZonesPage->new({app => YuiRestClient::get_app()});
+    $self->{WizardPage} = YaST::DNSServer::FinishWizardPage->new({app => YuiRestClient::get_app()});
+    $self->{WeakPasswordPopup} = Installation::Popups::YesNoPopup->new({app => YuiRestClient::get_app()});
+    return $self;
+}
+
+sub process_reading_configuration {
+    my ($self) = @_;
+    $self->continue_with_interfaces_controlled_by_nm();
+    return $self->get_forwarded_settings_page();
+}
+
+sub get_forwarded_settings_page {
```

Comment:
typo in several places, please check s/forwarded/forwarder/
That is why I would prefer to always call it the same, not sometimes fw, it is easy to mess up, this English work also for me is tricky :)